ConfigSpace/io/pcs_new.py

```diff
@@ -0,0 +1,452 @@
+            pass
+
+        try:
+            if "categorical" in line:
+                param_list = pp_cat_param.parseString(line)
+                name = param_list[0]
+                choices = [choice for choice in param_list[3:-4:2]]
+                default = param_list[-2]
+                param = create["categorical"](name=name, choices=choices, default=default)
+                cat_ct += 1
+                
+            elif "ordinal" in line:
+                param_list = pp_ord_param.parseString(line)
+                name = param_list[0]
+                sequence = [seq for seq in param_list[3:-4:2]]
+                default = param_list[-2]
+                param = create["ordinal"](name=name, sequence=sequence, default=default)
+                ord_ct += 1
+                
+        except pyparsing.ParseException:
+            pass
+
+        if param is None:
+            raise NotImplementedError("Could not parse: %s" % line)
+
+        configuration_space.add_hyperparameter(param)
+
+    for clause in forbidden:
+        param_list = pp_forbidden_clause.parseString(clause)
+        tmp_list = []
+        clause_list = []
+        for value in param_list[1:]:
+            if len(tmp_list) < 3:
+                tmp_list.append(value)
+            else:
+                # So far, only equals is supported by SMAC
+                if tmp_list[1] == '=':
+                    # TODO maybe add a check if the hyperparameter is
+                    # actually in the configuration space
+                    clause_list.append(ForbiddenEqualsClause(
+                        configuration_space.get_hyperparameter(tmp_list[0]),
+                        tmp_list[2]))
+                else:
+                    raise NotImplementedError()
+                tmp_list = []
+        configuration_space.add_forbidden_clause(ForbiddenAndConjunction(
+            *clause_list))
+
+
+    conditions_per_child = OrderedDict()
+    for condition in conditions:
+        child_name = condition[0]
+        if child_name not in conditions_per_child:
+            conditions_per_child[child_name] = list()
+        connectives = []
+        if condition[2] =='(':
+            i = 2
+            j = 2
+            while i < len(condition):
+                if condition[i] == '&&' or condition[i] == '||':
+                    connectives.append(condition[i])
+                    i += 1
+                if condition[i] == ')':
+                    conditions_per_child[child_name].append(condition[j:i+1])
+                    i +=1
+                    j = i+1
+                else:
+                    i += 1
+        else:
+            conditions_per_child[child_name].append(condition)
+    for child_name in conditions_per_child:
+        condition_objects = []
+        for condition in conditions_per_child[child_name]:
+            child = configuration_space.get_hyperparameter(child_name)
+            if '(' not in condition:
```

Comment:
Sorry Christina, I didn't know that SMAC 3 introduced yet another pcs format ... I, for once, liked the expressiveness of the SMAC 2.10 pcs format, but so be it ...